src/core/types/execution_context.ts

```diff
@@ -16,11 +16,14 @@ export type KibanaExecutionContext = {
   /**
    * Kibana application initated an operation.
    * */
-  readonly type: string; // 'visualization' | 'actions' | 'server' | ..;
-  /** public name of a user-facing feature */
-  readonly name: string; // 'TSVB' | 'Lens' |  'action_execution' | ..;
+  readonly type?: string; // 'visualization' | 'actions' | 'server' | ..;
+  /** public name of an application or a user-facing feature */
+  readonly name?: string; // 'TSVB' | 'Lens' |  'action_execution' | ..;
+  /** a stand alone, logical unit such as an application page or tab */
+  readonly page?: string; //
+  /** space identifier */
```

Comment:
Are we missing the space identifier property?

Reply:
It's complicated to add it (defined in xpack), so we'll probably add the space and the cloud org Id in the future